crates/fuel-streams-core/src/stream.rs

```diff
@@ -106,27 +104,36 @@ impl<S: Streamable> Stream<S> {
         &self,
         subject: &impl IntoSubject,
         payload: &S,
-    ) -> Result<u64, StreamError> {
+    ) -> Result<i64, StreamError> {
         let subject_name = &subject.parse();
-        self.store
-            .put(subject_name, payload.encode(subject_name).await.into())
-            .await
-            .map_err(|s| StreamError::PublishFailed {
+        let result = self
+            .store
+            .create(subject_name, payload.encode(subject_name).await.into())
```

Comment:
By using `store.create` instead `store.put` we ensure one subject will be added just once, and we can bypass the `AlreadyExists` to avoid crashing

Reply:
I dont feel this is the cleanest solution, but I agree that its fine to use it for the time being. Having our fuel fork of async_nats should not be an issue in my understanding. @luizstacio what do you reckon here ? Either way, I am fine with the change